ansible/roles/ci/unit_tests/tasks/main.yml

```diff
@@ -1,6 +1,32 @@
 ---
 # Playbook for unit tests
 
+- name: check if project_sources_path variable was set
+  fail: msg="Variable project_sources_path was not set"
+  when: project_sources_path is not defined
+
+- name: check if code_coverage_options variable was set
+  fail: msg="Variable code_coverage_options was not set"
+  when: code_coverage_options is not defined
+
+- name: Read all packages names in workspace
+  shell: find -type f -name package.xml -printf "%h\n" | xargs basename -a chdir={{ros_workspace}}
+  register: workspace_packages_list
+
+- name: Read all packages names in project
+  shell: find -type f -name package.xml -printf "%h\n" | xargs basename -a chdir={{project_sources_path}}
+  register: project_packages_list
+
+- name: Set blacklisted packages using lists substraction
```

Comment:
Use a more meaningful name? **Blacklist non-project packages** or something like this